lib/install.js

```diff
@@ -1,54 +1,44 @@
-var BinWrapper = require('bin-wrapper');
-var cleanInstallPath = require('./clean-install-path');
-var defaultConfig = require('./default-config');
-var noop = require('./noop');
-var spinner = require('simple-spinner');
-
-module.exports = function install(opts, cb) {
-  if (typeof opts === 'function') {
-    cb = opts;
-    opts = {};
-  }
-
-  if (typeof cb !== 'function') {
-    cb = noop;
-  }
-
-  if (typeof opts === 'undefined') {
-    opts = {};
-  }
-
-  var logger = require('./logger')(opts.logLevel);
-
-  var urlAndPath = defaultConfig.baseURL + '/v' + defaultConfig.version;
-
-  logger.silly('Theme Kit installation starting');
-  spinner.start();
-
-  var installer = new BinWrapper()
-    .src(urlAndPath + '/darwin-amd64/theme', 'darwin')
-    .src(urlAndPath + '/linux-386/theme', 'linux')
-    .src(urlAndPath + '/linux-amd64/theme', 'linux', 'x64')
-    .src(urlAndPath + '/windows-386/theme.exe', 'win32')
-    .src(urlAndPath + '/windows-amd64/theme.exe', 'win32', 'x64')
-    .dest(defaultConfig.destination)
-    .use(defaultConfig.binName);
-
-  var installPath = installer.path();
-
-  return cleanInstallPath(installPath, function(err) {
-    if (err) {
-      throw err;
-    }
-
-    return installer.run(['version'], function(runErr) {
+const BinWrapper = require('bin-wrapper');
+const spinner = require('simple-spinner');
+const config = require('./config');
+const {cleanFile} = require('./utils');
+
+/**
+ * Installs the Theme Kit executable into the bin/ subdirectory.
+ * Call this on npm postinstall.
+ * @param {string} logLevel   Log level
+ */
+function install(logLevel) {
+  return new Promise((resolve, reject) => {
+    const logger = require('./logger')(logLevel);
+    const urlAndPath = `${config.baseURL}/v${config.version}`;
+
+    logger.silly('Theme Kit installation starting');
+    spinner.start();
+
+    const installer = new BinWrapper()
+      .src(`${urlAndPath}/darwin-amd64/theme`, 'darwin')
+      .src(`${urlAndPath}/linux-386/theme`, 'linux')
+      .src(`${urlAndPath}/linux-amd64/theme`, 'linux', 'x64')
+      .src(`${urlAndPath}/windows-386/theme.exe`, 'win32')
+      .src(`${urlAndPath}/windows-amd64/theme.exe`, 'win32', 'x64')
+      .dest(config.destination)
+      .use(config.binName);
+
+    const pathToExecutable = installer.path();
+
+    cleanFile(pathToExecutable);
+
+    installer.run(['version'], (runErr) => {
```

Comment:
[BinWrapper](https://www.npmjs.com/package/bin-wrapper) returns a promise -- you should return this promise and ditch creating a `new Promise()`

Reply:
Unfortunately, due to a regression in the latest version of BinWrapper which promisified everything (https://github.com/kevva/bin-wrapper/issues/67), I think we're stuck with `3.0.2` and its callbacks until they resolve the issue.